src/data-fetching/api/PrezlyApi.ts

```diff
@@ -138,13 +143,18 @@ export class PrezlyApi {
         pinning = false,
         include,
         localeCode,
+        useHighlightedStory,
     }: GetStoriesOptions = {}) {
         const sortOrder = getChronologicalSortOrder(order, pinning);
         const query = JSON.stringify(getStoriesQuery(this.newsroomUuid, undefined, localeCode));
 
+        const limit = useHighlightedStory && (!page || page === 1) ? pageSize + 1 : pageSize;
+        const baseOffset = typeof page !== 'undefined' ? (page - 1) * pageSize : undefined;
+        const offset = useHighlightedStory && baseOffset ? baseOffset + 1 : baseOffset;
```

Comment:
There is currently a bug in the API, which does not allow to use `offset` that is not equally dividable by the `limit`. This is not how these parameters should work, so until this bug is resolved, this PR can't be merged.
See [QRF-699](https://linear.app/prezly/issue/QRF-699/sdkstoriessearch-api-endpoint-treats-offset-parameter-incorrectly)

Reply:
Let me fix the backend now. Then we don't need these hacks. 

Agree?

Reply:
This code is already made with API working as intended in mind. If we use the highlighted story, we need to offset each page by one.
E.g with page size of 10, without highlighted story, the pages would have offsets of 0, 10, 20, etc.
With highlighted story, since the first page is 11 stories, the offsets would be 0, 11, 21, etc

Reply:
Still, since the offset property does not work as expected, the backend fx still needed

Reply:
Ah, I see. Thanks :+1: 

One suggestion though. This code, if extracted to a function, would look much more readable (IMO):

```ts
function toPaginationParams(params: { page?: number, pageSize: number, withHighlightedStory: boolean }) {
    const { page = 1, pageSize, withHighlightedStory } = params;
    const offset = pageSize * (page - 1);
    const limit = pageSize;

    if (withHighlightedStory && page === 1) {
        return { offset: 0, limit: limit + 1 };
    }

    if (withHighlightedStory && page > 1) {
        return { offset: offset + 1, limit };
    }

    return { offset, limit };
}
```

Reply:
Yes, that looks much nicer! I'll address this and other comment in a bit.

Reply:
9d898aeafef3fed808b8011aa55608654fc897f5